packages/suite/src/actions/suite/bioAuthThunks.ts

```diff
@@ -1,176 +1,111 @@
-        ),
-    });
-
-const handleError = (error: string, dispatch: Dispatch, translationString: TranslationFunction) => {
-    dispatch(bioAuthActions.bioAuthValidated(null));
-
+const handleError = (error: string, dispatch: Dispatch, message: string) => {
     if (KNOWN_ERROR_MESSAGES.some(message => error.includes(message))) {
         // NOTE: known error message
         return;
     }
-
     dispatch(
         notificationsActions.addToast({
             type: 'error',
-            error: translationString('TR_BIO_AUTH_FAILED'),
+            error: message,
         }),
     );
 };
 
+export const init = createThunk(`${BIO_AUTH_PREFIX}/init`, async (_args, { dispatch }) => {
+    // settings
+    desktopApi.getBioAuthSettings().then(settings => {
+        console.log('get settings', settings);
+        dispatch(bioAuthActions.setBioAuthEnabled(settings.enabled));
+    });
+    desktopApi.on('bio-auth/settings-changed', settings => {
+        console.log('bio-auth/settings-changed', settings);
+        dispatch(bioAuthActions.setBioAuthEnabled(settings.enabled));
+    });
+
+    // api availability
+    desktopApi.isBioAuthAvailable().then(available => {
+        console.log('initial availability', available);
+        dispatch(bioAuthActions.setIsBioAuthAvailable(available));
+    });
+
+    // validation status
+    desktopApi.getBioAuthStatus().then(validated => {
+        console.log('initial validation status', validated);
+        dispatch(bioAuthActions.setIsBioAuthValidationRequired(!validated));
+    });
+    desktopApi.on('bio-auth/validation-status-changed', validated => {
+        console.log('validation status changed', validated);
+        dispatch(bioAuthActions.setIsBioAuthValidationRequired(!validated));
+    });
+    desktopApi.on('bio-auth/bio-auth-availability-changed', available => {
+        console.log('bio-auth availability changed', available);
+        dispatch(bioAuthActions.setIsBioAuthAvailable(available));
+    });
+});
+
 export const requestBioAuthChangeThunk = createThunk(
     `${BIO_AUTH_PREFIX}/requestBioAuthChangeThunk`,
     async (
         {
-            nextBioAuthEnabledValue,
-            translationString,
+            payload,
+            messageSuccess,
+            messageError,
         }: {
-            translationString: TranslationFunction;
-            nextBioAuthEnabledValue?: boolean;
+            payload: boolean;
+            messageSuccess: string;
+            messageError: string;
         },
-        { dispatch, getState },
+        { dispatch },
     ) => {
-        const prevBioEnabled = selectBioAuthEnabled(getState());
-        const isRequestingChange = selectIsRequestingBioAuthChange(getState());
-        const nextBioEnabled = nextBioAuthEnabledValue ?? !prevBioEnabled;
-
-        if (nextBioEnabled === prevBioEnabled) {
+        if (!(await desktopApi.isBioAuthAvailable())) {
+            dispatch(notificationsActions.addToast({ type: 'error', error: 'not available' }));
```

Comment:
Make it more understandable for the end-user 🙏 
```suggestion
            dispatch(notificationsActions.addToast({ type: 'error', error: 'Biometric authentication not available' }));
```

Reply:
f6783e985816ca218749ab622948eba909f3cd96